source/Ship.cpp

```diff
@@ -2043,68 +2096,115 @@ void Ship::Move(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flotsam)
 		acceleration = Point();
 	}
 
+	static const double weaponsRangeMultiplier = 1.25;
+	bool hasPrimary = firePrimary;
+
 	// Boarding:
 	shared_ptr<const Ship> target = GetTargetShip();
+
 	// If this is a fighter or drone and it is not assisting someone at the
 	// moment, its boarding target should be its parent ship.
 	if(CanBeCarried() && !(target && target == GetShipToAssist()))
 		target = GetParent();
-	if(target && !isDisabled)
+	if(!isDisabled)
 	{
-		Point dp = (target->position - position);
-		double distance = dp.Length();
-		Point dv = (target->velocity - velocity);
-		double speed = dv.Length();
-		isBoarding = (distance < 50. && speed < 1. && commands.Has(Command::BOARD));
-		if(isBoarding && !CanBeCarried())
-		{
-			if(!target->IsDisabled() && government->IsEnemy(target->government))
-				isBoarding = false;
-			else if(target->IsDestroyed() || target->IsLanding() || target->IsHyperspacing()
-					|| target->GetSystem() != GetSystem())
-				isBoarding = false;
-		}
-		if(isBoarding && !pilotError)
+		if(target)
 		{
-			Angle facing = angle;
-			bool left = target->Unit().Cross(facing.Unit()) < 0.;
-			double turn = left - !left;
-
-			// Check if the ship will still be pointing to the same side of the target
-			// angle if it turns by this amount.
-			facing += TurnRate() * turn;
-			bool stillLeft = target->Unit().Cross(facing.Unit()) < 0.;
-			if(left != stillLeft)
-				turn = 0.;
-			angle += TurnRate() * turn;
-
-			velocity += dv.Unit() * .1;
-			position += dp.Unit() * .5;
-
-			if(distance < 10. && speed < 1. && (CanBeCarried() || !turn))
+			Point dp = (target->position - position);
+			double distance = dp.Length();
+			Point dv = (target->velocity - velocity);
+			double speed = dv.Length();
+			isBoarding = (distance < 50. && speed < 1. && commands.Has(Command::BOARD));
+
+			bool activeEnemyTarget = !target->IsDisabled() && government->IsEnemy(target->government);
+
+			if(!commands.Has(Command::JUMP) && !hasPrimary)
 			{
-				if(cloak)
+				bool targetInRange = target->Position().Distance(this->Position()) < weaponsRangeMultiplier * this->weaponRange 
+									|| this->weaponRange == 0.0;
```

Comment:
```suggestion
									|| !this->weaponRange;
```
!variable is the same as variable == 0